plugins/omelasticsearch/omelasticsearch.c

```diff
@@ -1084,14 +1151,14 @@ static rsRetVal
 curlPost(wrkrInstanceData_t *pWrkrData, uchar *message, int msglen, uchar **tpls, int nmsgs)
 {
 	CURLcode code;
-	CURL *curl = pWrkrData->curlHandle;
+	CURL *curl = pWrkrData->curlPostHandle;
 	DEFiRet;
 
 	pWrkrData->reply = NULL;
 	pWrkrData->replyLen = 0;
 
-	if(pWrkrData->pData->dynSrchIdx || pWrkrData->pData->dynSrchType || pWrkrData->pData->dynParent)
-		CHKiRet(setCurlURL(pWrkrData, pWrkrData->pData, tpls));
+	CHKiRet(checkConn(pWrkrData));
```

Comment:
The individual servers won't have backoff, only in the state that all servers are inaccessible will the core functionality happen. I have it calling checkConn before every POST to ES, which cycles through all servers until it finds one that works. Given how ES works, the worst case scenario (with partial cluster availability) is that we spend `(N-1)/2 * healthCheckTimeout` every time we submit to ES.